src/Compilers/CSharp/Test/Semantic/Semantics/InlineArrayTests.cs

```diff
@@ -4241,6 +4263,36 @@ static async Task<T> FromResult<T>(T r)
                 );
         }
 
+        [Fact]
+        public void ElementAccess_ExpressionTree_01()
+        {
+            var src = @"
+using System.Linq.Expressions;
+
+class Program
+{
+    static Expression<System.Func<int>> M1(Buffer10<int> x) =>
+        () => x[0];
+
+    static Expression<System.Action> M2(Buffer10<int> x) =>
+        () => x[0] = 111;
+}
+" + Buffer10Definition;
+
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.ReleaseDll);
+            comp.VerifyDiagnostics(
+                // (7,15): error CS9507: An expression tree may not contain an inline array access or conversion
+                //         () => x[0];
+                Diagnostic(ErrorCode.ERR_ExpressionTreeContainsInlineArrayOperation, "x[0]").WithLocation(7, 15),
+                // (10,15): error CS0832: An expression tree may not contain an assignment operator
+                //         () => x[0] = 111;
+                Diagnostic(ErrorCode.ERR_ExpressionTreeContainsAssignment, "x[0] = 111").WithLocation(10, 15),
+                // (10,15): error CS9507: An expression tree may not contain an inline array access or conversion
+                //         () => x[0] = 111;
+                Diagnostic(ErrorCode.ERR_ExpressionTreeContainsInlineArrayOperation, "x[0]").WithLocation(10, 15)
+                );
+        }
```

Comment:
Are we testing the conversion case? "An expression tree may not contain an inline array [...] conversion"